gratipay/models/participant/__init__.py

```diff
@@ -1135,8 +1135,11 @@ def change_username(self, suggested):
 
         lowercased = suggested.lower()
 
-        if lowercased in gratipay.RESTRICTED_USERNAMES:
-            raise UsernameIsRestricted(suggested)
+        # Don't allow any username which is the name of a
+        # file existing on the web_root folder.
+        for ext in ['', '.spt', '.ico', '.txt', '.osdd']:
+            if (lowercased + ext) in gratipay.RESTRICTED_USERNAMES:
+                raise UsernameIsRestricted(suggested)
```

Comment:
A redirect (to either `/username/` or `/~username`) would indicate that the username is available. A 200 would indicate that it is not.